nall/function.hpp

```diff
@@ -37,12 +36,6 @@ template<typename R, typename... P> struct function<auto (P...) -> R> {
     return *this;
   }
 
-  auto operator=(void* source) -> function& {
```

Comment:
Clang couldn't decide which assignment operator to use when this one was an option. Casting function pointers to void* is implementation-defined and kind of frowned upon anyway, so I decided to just drop it.